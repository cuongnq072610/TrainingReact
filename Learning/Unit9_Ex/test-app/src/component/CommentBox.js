import React from 'react'

export default (props) => (
  <div>
    <form>
      <h4>Comment Box</h4>
      <textarea value={props.value}/>
      <div><button>Send Comment</button></div>
    </form>
  </div>
)