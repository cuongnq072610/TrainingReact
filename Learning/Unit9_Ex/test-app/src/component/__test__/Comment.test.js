import React from 'react'
import CommentBox from '../CommentBox'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer';
let wrapped
// Khởi tạo cho tất cả các testcase
// beforeEach(() => {
    
// })

// afterEach(() => {
    
// })

it('has a text area and a button', () => {
    wrapped = mount(<CommentBox />)
    expect(wrapped.find('textarea').length).toEqual(1)
    expect(wrapped.find('button').length).toEqual(1)
    wrapped.unmount()
})

//Snapshot Example to check UI 
it('render correctly', () => {
    wrapped = renderer.create(<CommentBox />).toJSON()
    expect((wrapped)).toMatchSnapshot()
})