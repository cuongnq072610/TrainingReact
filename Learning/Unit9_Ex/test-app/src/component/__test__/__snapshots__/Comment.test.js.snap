// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div>
  <form>
    <h4>
      Comment Box
    </h4>
    <textarea />
    <div>
      <button>
        Send Comment
      </button>
    </div>
  </form>
</div>
`;
